cdap-ui/app/features/admin/controllers/NamespaceDatasetMetadataController.js

```diff
@@ -1,15 +1,27 @@
 angular.module(PKG.name + '.feature.admin').controller('AdminNamespaceDatasetMetadataController',
```

Comment:
Ok. In 3.1 maybe, we should change that, right?
When making a request, we should have some flexibility in allowing 404s, and handling them appropriately in the controller ourself.